### PR TITLE
Friend model changes, New API, Friend Request Component changes

### PR DESCRIPTION
**Pull request by taylorfinelli**
# What type of PR is this? (Check all that apply)

- [x] ✨**feature**: Introduces completely new code or new features.
- [ ] 🐛**fix**: Implements changes that fix a bug. Ideally, reference an issue if present.
- [ ] ♻️**refactor**: Includes any code-related change that is neither a fix nor a feature.
- [ ] ✅**build**: Encompasses all changes related to the build of the software, including changes to dependencies or the addition of new ones.
- [ ] ⚡️**test**: Pertains to all changes regarding tests, whether adding new tests or modifying existing ones.
- [ ] 🚰**ci**: Involves all changes related to the configuration of continuous integration, such as GitHub Actions or other CI systems.
- [ ] 📚**docs**: Includes all changes to documentation, such as README files, or any other documentation present in the repository.
- [ ] 🗑️**chore**: Captures all changes to the repository that do not fit into the above categories.

# Description

- Modified friend model in schema
  - Now has `friendId` that can be used to query their friend info
- Modified the `friend.create` API -> `makeFriendsReceiverIdSenderId`
  - Makes a two-way friendship between the two users
  - Deletes the friend request notification
  - Implemented this API in the frontend
- Changes to context to make the data query properly  

# Tests

How was this tested?

- [ ] Unit tests
- [ ] Integration tests
- [ ] E2E tests
- [x] Manual tests
- [ ] Tests were NOT needed
- [ ] Other (explain below)

# [Optional] Screenshots
https://github.com/UNLV-CS472-672/2024-S-GROUP3-Barbell/assets/85365584/599da941-323b-4751-9bf1-092b98b1a090

# Documentation

- [ ] Added to README.me
- [ ] Seperate document
- [x] NO documentation needed

### CODE REVIEW

File: packages/db/src/mock-data/friend.json
Comment (mva919) [inline packages/db/src/mock-data/friend.json:1]: ?

File: packages/db/src/mock-data/friend.json
Comment (taylorfinelli) [inline packages/db/src/mock-data/friend.json:1]: I deleted this because it uses a relationship and we can't set those in the mock data i think

File: packages/api/src/router/friend.ts
Comment (mva919) [inline packages/api/src/router/friend.ts:40]: NITPICK: Could make this an int as well

File: packages/db/src/mock-data/friend.json
Comment (mva919) [reply packages/db/src/mock-data/friend.json:1]: Oh I SEEE

File: packages/api/src/router/friend.ts
Comment (taylorfinelli) [reply packages/api/src/router/friend.ts]: Ok, done

Comment (taylorfinelli) [reply packages/api/src/router/friend.ts]: Thanks

File: packages/db/src/mock-data/friend.json
Comment (taylorfinelli) [reply packages/db/src/mock-data/friend.json:1]: Yeah but I'm not sure because sometimes they're set whenever we seed so maybe we could look into it

**Review by mva919 — APPROVED** [review]
Looks good, we should get someone else's approval before merging since I touched the global context 😈

**Comment by taylorfinelli** [comment]
assigning @thienguen

File: packages/db/src/mock-data/friend.json
Comment (mva919) [reply packages/db/src/mock-data/friend.json:1]: hmm okay, yea it would be nice to have the relationships with the seeds

File: packages/db/src/mock-data/friend.json
Comment (thienguen) [reply packages/db/src/mock-data/friend.json:1]: why

File: packages/db/src/mock-data/friend.json
Comment (thienguen) [reply packages/db/src/mock-data/friend.json:1]: ye we could do that, issue opened

**Review by thienguen — APPROVED** [review]

**Comment by thienguen** [comment]
approved

**Pull request merged**
